portfolio/src/main/webapp/mapScript.js

```diff
@@ -0,0 +1,155 @@
+        let max = $('#quantity').val();
+        getComments(max);
+    });
+};
+
+const deleteAllComments = function() {
+    const request = new Request('/delete-data', {method: 'POST'});
+    fetch(request).then(response => {
+        getComments();
+    });
+};
+
+const handleDeleteButton = function() {
+    $('#delete-button').click(function() {
+        deleteAllComments();
+    });
+};
+
+let map;
+
+/* Editable marker that displays when a user clicks in the map. */
+let editMarker;
+
+const initMap = function() {
+    const unc = {lat: 35.905112, lng: -79.046892}
+    map = new google.maps.Map(document.getElementById("map"), {
+        center: unc,
+        zoom: 14,
+    });
+
+    map.addListener('click', (event) => {
+        createMarkerForEdit(event.latLng.lat(), event.latLng.lng());
+    });
+    /** Fetches markers from datastore */
+    fetch('/markers').then(response => response.json()).then((markers) => {
+        for (let marker of markers) {
+            createMarkerForDisplay(marker.lat, marker.lng, marker.title, marker.description);
+        }
+    });
+}
+
+
+/** Creates a marker that shows a read-only info window when clicked. */
+const createMarkerForDisplay = function(lat, lng, title, description) {
+    const marker = new google.maps.Marker({position: {lat: lat, lng: lng}, map: map, title: title});
+    const content = '<div><h1>'+ title +'</h1></div>' + 
+                    '<div><p>' + description + '</p></div>';
+    $('#place-list').append('<li>' + marker.title + '</li>');
+    const infoWindow = new google.maps.InfoWindow({content: content});
+    marker.addListener('click', () => {
+        infoWindow.open(map, marker);
+    });
+}
+
+/** Sends a marker to the backend for saving. */
+const postMarker = function(lat, lng, title, description) {
+    const params = new URLSearchParams();
+    params.append('lat', lat);
+    params.append('lng', lng);
+    params.append('title', title);
+    params.append('description', description);
+   
+    fetch('/markers', {method: 'POST', body: params});
+}
+
+/** Creates a marker that shows a textbox the user can edit. */
+const createMarkerForEdit = function(lat, lng) {
+    // If we're already showing an editable marker, then remove it.
+    if (editMarker) {
+        editMarker.setMap(null);
+    }
+
+    editMarker = new google.maps.Marker({position: {lat: lat, lng: lng}, map: map});
+    const infoWindow = new google.maps.InfoWindow({content: buildInfoWindowInput(lat, lng)});
+
+    // When the user closes the editable info window, remove the marker.
+    google.maps.event.addListener(infoWindow, 'closeclick', () => {
+        editMarker.setMap(null);
+    });
+    infoWindow.open(map, editMarker);
+}
+
+/**
+ * Builds and returns HTML elements that show an editable textbox and a submit
+ * button.
+ */
+const buildInfoWindowInput = function(lat, lng) {
+    const title = document.createElement('input');
+    title.setAttribute("placeholder", "Title")
```

Comment:
According to the Google JS style guide, use single quotes for strings (here and elsewhere in the file)